internal/tfprotov5tov6/tfprotov5tov6.go

```diff
@@ -1036,18 +1036,17 @@ func ListResourceServerStream(in *tfprotov5.ListResourceServerStream) *tfprotov6
 	return &tfprotov6.ListResourceServerStream{
 		Results: func(yield func(tfprotov6.ListResourceResult) bool) {
 			for res := range in.Results {
-				yield(*ListResourceResult(&res))
+				if !yield(ListResourceResult(res)) {
+					break
+				}
+				yield(ListResourceResult(res))
```

Comment:
Since we called `yield()` above, this would call `yield()` a second time. Let's remove the second `yield()`.